poller.go

```diff
@@ -254,6 +255,7 @@ func (p *Poller) Wait(ctx context.Context, interval time.Duration) <-chan error
 									successful = 0
 									continue
 								}
+								metricChan <- fmt.Sprintf("initial value %g, current value %g (%s elapsed)", initVal, resVal, p.timer.Elapsed())
```

Comment:
I think we may want to write to this channel in a non-blocking way. Otherwise if nothing is waiting for something to be read, it'll block. 

Maybe something like:
```go
select {
  case metric <- fmt.Sprintf("initial value %g, current value %g (%s elapsed)", initVal, resVal, p.timer.Elapsed()):
  default:
}
```

Reply:
Great point, thanks.

Reply:
So when I made this change, (along with two other changes, adding a default case to the for-select following `p.Wait` and changing `break` to `return nil`) it appeared that the `metricChan <- msg` case was skipped. Wouldn't that suggest `metricChan` wasn't open? On other other hand, if I leave it as is (i.e. blocking) it worked as intended (every interval a printed an updated).

Reply:
I pushed those changes, so at least the command manually works as intended.